hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java

```diff
@@ -819,4 +832,85 @@ private void connectToFeedCommChannel() throws Exception {
         feedComm = new FeedComm(this.httpClientBuilder, this.configuration, this.feedId, this.dmrServerInventories);
         feedComm.connect();
     }
+
+    private static final class ServerAddressProvider implements InventoryStorage {
+        private final InventoryStorage delegate;
+        private final ManagedServer server;
+
+        public ServerAddressProvider(ManagedServer server, InventoryStorage delegate) throws UnknownHostException {
+            super();
+            this.delegate = delegate;
+            this.server = server;
+        }
+
+        private InetAddress[] resolveHost() throws UnknownHostException {
+            String host = null;
+            if (server instanceof RemoteDMRManagedServer) {
+                RemoteDMRManagedServer remoteServer = (RemoteDMRManagedServer) server;
+                host = remoteServer.getHost();
+            } else if (server instanceof LocalDMRManagedServer) {
+                host = InetAddress.getLocalHost().getCanonicalHostName();
+            } else {
+                throw new IllegalStateException("Unexpected type of managed server '" + server.getClass().getName()
+                        + "'; expected '" + RemoteDMRManagedServer.class.getName() + "' or '"
+                        + LocalDMRManagedServer.class.getName() + "'. Please report this bug.");
+            }
+            MsgLogger.LOG.tracef("About to resolve IP addresses for host [%s]", host);
+            return InetAddress.getAllByName(host);
+        }
+
+        @Override
+        public void storeResourceType(ResourceType<?, ?, ?, ?> resourceType) {
+            delegate.storeResourceType(resourceType);
+        }
+
+        @Override
+        public void storeResource(Resource<?, ?, ?, ?, ?> resource) {
+            final String WF_TYPE_NAME = "WildFly Server";
```

Comment:
Isn't that something for a agent-wide constant, that is defined once?
Also suppose that in the future layered stuff would be deployed on top of WF - e.g. Hawkular, where the type could be "Hawkular Server <extends> WildFly Server", how would that work together?